core/services/fluxmonitorv2/flux_monitor_test.go

```diff
@@ -0,0 +1,1674 @@
+	}, nil).Once().
+		Run(func(mock.Arguments) { close(chRoundState2) }).
+		Once()
+
+	fm, err := fluxmonitorv2.NewFluxMonitor(
+		NewPipelineRun(),
+		fluxmonitorv2.NewStore(store.DB, store, jobORM, pipelineORM),
+		pollTicker,
+		fluxmonitorv2.NewPaymentChecker(assets.NewLink(1), nil),
+		spec.ContractAddress,
+		fluxmonitorv2.NewFluxAggregatorContractSubmitter(fluxAggregator, transmissionAddress),
+		fluxmonitorv2.NewDeviationChecker(spec.Threshold, spec.AbsoluteThreshold),
+		fluxmonitorv2.Flags{},
+		fluxAggregator,
+		logBroadcaster,
+		spec,
+		func() {},
+		big.NewInt(0),
+		big.NewInt(100000000000),
+	)
+	require.NoError(t, err)
+
+	fm.Start()
+	fm.OnConnect()
+
+	logBroadcast := new(logmocks.Broadcast)
+	logBroadcast.On("WasAlreadyConsumed").Return(false, nil)
+	logBroadcast.On("DecodedLog").Return(&flux_aggregator_wrapper.FluxAggregatorNewRound{RoundId: big.NewInt(0), StartedAt: big.NewInt(time.Now().UTC().Unix())})
+	logBroadcast.On("MarkConsumed").Return(nil)
+	fm.HandleLog(logBroadcast, nil)
+
+	gomega.NewGomegaWithT(t).Eventually(chRoundState1).Should(gomega.BeClosed())
+	gomega.NewGomegaWithT(t).Eventually(chRoundState2).Should(gomega.BeClosed())
+
+	time.Sleep(time.Duration(2*timeout) * time.Second)
+	fm.Close()
+
+	fluxAggregator.AssertExpectations(t)
+}
+
+func TestFluxMonitor_SufficientFunds(t *testing.T) {
+	store, cleanup := cltest.NewStore(t)
+	defer cleanup()
+
+	var (
+		fluxAggregator = new(mocks.FluxAggregator)
+		logBroadcaster = new(logmocks.Broadcaster)
+		jobORM         = new(jobmocks.ORM)
+		pipelineORM    = new(pipelinemocks.ORM)
+		spec           = NewSpecification()
+		pollTicker     = fluxmonitorv2.NewPollTicker(time.Minute, false)
+	)
+
+	checker, err := fluxmonitorv2.NewFluxMonitor(
+		NewPipelineRun(),
+		fluxmonitorv2.NewStore(store.DB, store, jobORM, pipelineORM),
+		pollTicker,
+		fluxmonitorv2.NewPaymentChecker(assets.NewLink(1), nil),
+		spec.ContractAddress,
+		fluxmonitorv2.NewFluxAggregatorContractSubmitter(fluxAggregator, transmissionAddress),
+		fluxmonitorv2.NewDeviationChecker(spec.Threshold, spec.AbsoluteThreshold),
+		fluxmonitorv2.Flags{},
+		fluxAggregator,
+		logBroadcaster,
+		spec,
+		func() {},
+		big.NewInt(0),
+		big.NewInt(100000000000),
+	)
+	require.NoError(t, err)
+
+	payment := 100
+	rounds := 3
+	oracleCount := 21
+	min := payment * rounds * oracleCount
+
+	testCases := []struct {
+		name  string
+		funds int
+		want  bool
+	}{
+		{"above minimum", min + 1, true},
+		{"equal to minimum", min, true},
+		{"below minimum", min - 1, false},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			state := flux_aggregator_wrapper.OracleRoundState{
+				AvailableFunds: big.NewInt(int64(tc.funds)),
+				PaymentAmount:  big.NewInt(int64(payment)),
+				OracleCount:    uint8(oracleCount),
+			}
+			assert.Equal(t, tc.want, checker.ExportedSufficientFunds(state))
+		})
+	}
+}
+
+func TestFluxMonitor_SufficientPayment(t *testing.T) {
```

Comment:
This now delegates to the PaymentChecker but I've left this in for sanity checks for now.

Reply:
I now call the `PaymentChecker.SufficientPayment` directly and have removed this test